src/ui/Alert/Alert.tsx

```diff
@@ -0,0 +1,104 @@
+import { cva, VariantProps } from 'cva'
+import * as React from 'react'
+
+import { cn } from 'shared/utils/cn'
+
+import successImage from './assets/check-circle.svg'
+import warningImage from './assets/exclamation.svg'
+import infoImage from './assets/information-circle.svg'
+import errorImage from './assets/x-circle.svg'
+
+const alertVariants = cva(
+  '[&>svg]:text-foreground relative w-full border-l-4 p-4 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg~*]:pl-7',
+  {
+    variants: {
+      variant: {
+        error: 'border-ds-primary-red bg-error-100',
+        info: 'border-ds-blue-darker bg-ds-blue-nonary',
+        success: 'border-green-500 bg-green-100',
+        warning: 'border-orange-500 bg-orange-100',
+      },
+    },
+    defaultVariants: {
+      variant: 'info',
+    },
+  }
+)
+
+export function variantToIcon(variant?: string | null) {
+  let alt = 'information circle'
+  let src = infoImage
+  switch (variant) {
+    case 'error':
+      alt = 'error'
+      src = errorImage
+      break
+    case 'info':
+      alt = 'information circle'
+      src = infoImage
+      break
+    case 'success':
+      alt = 'checkmark in circle'
+      src = successImage
+      break
+    case 'warning':
+      alt = 'triangle exclamation'
+      src = warningImage
+      break
+  }
+  return (
+    <img className="float-left -mt-1 mr-2 align-middle" src={src} alt={alt} />
+  )
+}
+
+interface RootProps
+  extends React.HTMLAttributes<HTMLHeadingElement>,
+    VariantProps<typeof alertVariants> {}
+
+const AlertRoot = React.forwardRef<HTMLDivElement, RootProps>(
+  ({ className, variant, ...props }, ref) => (
+    <div
+      ref={ref}
+      role="alert"
+      className={cn(alertVariants({ variant }), className)}
+      {...props}
+    >
+      {variantToIcon(variant)}
+      {props.children}
+    </div>
+  )
+)
+AlertRoot.displayName = 'Alert'
+
+const title = cva(['mb-1 font-semibold leading-none tracking-tight'])
+
+interface TitleProps
+  extends React.HTMLAttributes<HTMLHeadingElement>,
+    VariantProps<typeof title> {}
+
+const Title = React.forwardRef<HTMLParagraphElement, TitleProps>(
+  ({ className, ...props }, ref) => (
+    // eslint-disable-next-line jsx-a11y/heading-has-content
+    <h5 ref={ref} className={cn(title({ className }))} {...props} />
+  )
+)
+Title.displayName = 'Alert.Title'
+
+const description = cva(['text-sm [&_p]:leading-relaxed'])
+
+interface DescriptionProps
+  extends React.HTMLAttributes<HTMLHeadingElement>,
+    VariantProps<typeof description> {}
+
+const Description = React.forwardRef<HTMLParagraphElement, DescriptionProps>(
+  ({ className, ...props }, ref) => (
+    <div ref={ref} className={cn(description({ className }))} {...props} />
```

Comment:
You have this as a `HTMLParagraphElement` but the actual element is a `div`, not a `p`. This could lead to issues as the caller will assume it's a `p`. Is there a reason you did this? Might be a workaround

Reply:
tbh with you I just copied this over from the shadcn manual code. They also have a paragraphElement with the div here: https://ui.shadcn.com/docs/components/alert

admittedly, I haven't really looked into these types too much

Reply:
Oh! Well if they're doing it, it's probably fine to leave as is. If it ain't broke for Vercel...

Reply:
The alternative is we just make it a `p` element

Reply:
yea @nicholas-codecov

Reply:
Easy enough! Changed in [d874761](https://github.com/codecov/gazebo/pull/2947/commits/d8747618bf054230f4c5dea6ef1189750d03a24d)